gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java

```diff
@@ -65,8 +70,31 @@
   private SingleSelectDialogFactory singleSelectDialogFactory;
   private MultiSelectDialogFactory multiSelectDialogFactory;
 
-  @Nullable private EditObservationBottomSheetBinding addPhotoBottomSheetBinding;
-  @Nullable private BottomSheetDialog bottomSheetDialog;
+  private static Class<? extends AbstractFieldViewModel> getViewModelClass(Field.Type fieldType) {
```

Comment:
Rather than have `getViewModelClass` and `getFieldLayoutId`, can we instead have a `FieldViewFactory` class that will inflate the appropriate view and viewmodel? This fragment can coordinate the returned views by getting their viewmodels with `DataBindingUtil`. Wdyt?

Reply:
Great idea! d4a40af